Google.Api.Gax.Grpc/Rest/HttpRuleTranscoder.cs

```diff
@@ -268,12 +270,44 @@ internal IEnumerable<string> GetValues(IMessage request)
                 // uint32, fixed32, uint64, fixed64
                 value is UnsignedInt32Zero || value is 0UL || 
                 // bytes
-                (value is ByteString bs && bs.IsEmpty);
+                (value is ByteString bs && bs.IsEmpty) ||
+                // Any enum (all protobuf enums have an underlying type of Int32)
+                (value is Enum enumValue && Convert.ToInt32(enumValue) == 0);
 
             static string FormatValue(object value) => 
                 value is bool b ? (b ? "true" : "false")
+                : value is Enum enumValue ? OriginalEnumValueHelper.GetOriginalName(enumValue)
                 : value is IFormattable formattable ? formattable.ToString(format: null, CultureInfo.InvariantCulture)
                 : value.ToString();
         }
+
+        // Effectively a cache of mapping from enum values to the original name as specified in the proto file,
+        // fetched by reflection. This code is taken from the protobuf JsonFormatter.
```

Comment:
I'm guessing the protobuf JsonFormatter is private then, or this code is at least?

Reply:
This code is private, yes. It might be good to expose it at some point, but that can wait.